src/eradiate/test_tools/regression.py

```diff
@@ -62,17 +75,50 @@ def regression_test_plots(
 
     axes[0][1].set_axis_off()
     axes[0][1].legend(handles=handles, labels=labels, loc="upper center")
-    axes[0][1].text(
-        0.5, 0.5, f"{metric[0]}: {metric[1]:.4}", horizontalalignment="center"
-    )
+
+    if metric[1] is None:
+        axes[0][1].text(
+            0.5,
+            0.5,
+            f'Metric "{metric[0]}" is not available',
+            horizontalalignment="center",
+        )
+    else:
+        axes[0][1].text(
+            0.5, 0.5, f"{metric[0]}: {metric[1]:.4}", horizontalalignment="center"
+        )
 
     plt.tight_layout()
-    plt.savefig(filename)
-    plt.close()
+
+
+def render_svg_chart():
+    str_i = StringIO()
+    plt.savefig(str_i, format="svg", transparent=True)
+    svg = str_i.getvalue()
+
+    # Include some CSS in the SVG to render nicely in Robot report's dark and
+    # light modes
+    return "\n".join(
+        [
+            "<svg",
+            'version="1.1"',
+            'baseProfile="full"',
+            'width="810" height="540" viewBox="0 0 810 540"'
+            'xmlns="http://www.w3.org/2000/svg">',
+            "<style>",
+            "    path {",
+            "        fill: var(--text-color);",
+            "        stroke: var(--text-color);",
+            "    }",
+            "</style>",
+            svg,
+            "</svg>",
+        ]
+    )
 
 
 def reference_converter(
-    value: os.PathLike | xr.Dataset | None,
+    value: os.PathLike | xr.Dataset | str | bytes | None,
```

Comment:
We have a `PathLike`  type alias in the `eradiate.typing` module to make this a little more concise:

```suggestion
    value: PathLike | xr.Dataset | None,
```